common/djangoapps/course_about/data.py

```diff
@@ -1,18 +1,55 @@
-"""
-Data Aggregation Layer for the Course About API. All knowledge of edx-platform specific data structures should
-be hidden behind this layer.  The Python API (api.py) will access all data directly through this module.
-
+"""Data Aggregation Layer for the Course About API.
 This is responsible for combining data from the following resources:
 * CourseDescriptor
 * CourseAboutDescriptor
+"""
+from opaque_keys import InvalidKeyError
+from course_about.serializers import serialize_content
+from course_about.errors import CourseNotFoundError
+from lms.djangoapps.courseware import courses
+import logging
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
 
-Eventually, additional Marketing metadata will also be accessed through this layer.
+log = logging.getLogger(__name__)
 
-"""
+ABOUT_ATTRIBUTES = [
+    'effort',
+    'video'
+]
 
 
-def get_course_info_details(course_id):  # pylint: disable=unused-argument
-    """Return all relative course metadata
+def get_course_about_details(course_id):  # pylint: disable=unused-argument
+    """
+    Return course information for a given course id.
+    Args:
+        course_id : The course id to retrieve course information for.
+    Returns:
+        Serializable dictionary of the Course About Information.
+    Raises:
+        CourseNotFoundError
+    """
+    try:
+        course_key = CourseKey.from_string(course_id)
+        course_descriptor = courses.get_course(course_key, depth=0)
+    except InvalidKeyError as err:
+        raise CourseNotFoundError(err.message)
+    except ValueError as err:
+        raise CourseNotFoundError(err.message)
+    about_descriptor = {}
+    for attribute in ABOUT_ATTRIBUTES:
+        about_descriptor[attribute] = _fetch_course_detail(course_key, attribute)
+    return serialize_content(course_descriptor=course_descriptor, about_descriptor=about_descriptor)
 
+
+def _fetch_course_detail(course_key, attribute):
+    """
+    Fetch the course about attribute for the given course's attribute from persistence and return its value.
     """
-    pass
+    usage_key = course_key.make_usage_key('about', attribute)
+    try:
+        value = modulestore().get_item(usage_key).data
```

Comment:
I'll see if I have some time to investigate this further, but for now you can leave it as is and we can revisit this later if we need to.